flexmeasures/data/services/scheduling.py

```diff
@@ -65,6 +72,8 @@ def create_scheduling_job(
     )
     if enqueue:
         current_app.queues["scheduling"].enqueue_job(job)
+
+    scheduler.persist_flex_model()
```

Comment:
Would it make sense if `create_scheduling_job` and `make_schedule` receive a `Scheduler` instance instead of `scheduler_kwargs`? Then the `create_scheduling_job` can stay more focused. Also, both functions can then check themselves whether the flex config has already been deserialized, and it's one way of preventing the entanglement of flex state persistence with job creation.

Reply:
Make_schedule should get serializable arguments, as it might be queued via RQ

Reply:
I think so. So then it would be up to both the CLI and API to make the call to `persist_flex_model`, right?

Reply:
If it happens in make_schedule they don't have to do it

Reply:
I see. So then the flex state would be persisted when the job is executed rather than when it is set up. I agree that it is better than the current place, where persisting the flex state would be missed when the schedule is created directly. I also feel it goes some way towards resolving my initial issue, because then persisting the flex state would be positioned next to where the schedules are persisted, and where I'd also envision persisting the scheduled SoC in the near future.